sdk-lib/src/main/java/cash/z/ecc/android/sdk/jni/Backend.kt

```diff
@@ -97,9 +97,10 @@ internal interface Backend {
     suspend fun rewindBlockMetadataToHeight(height: Long)
 
     /**
+     * @param limit The limit provides an efficient way how to restrict the portion of blocks, which will be validated.
      * @return Null if successful. If an error occurs, the height will be the blockheight where the error was detected.
      */
-    suspend fun validateCombinedChainOrErrorHeight(): Long?
+    suspend fun validateCombinedChainOrErrorHeight(limit: Int = -1): Long?
```

Comment:
Yes, I agree on the Long type instead of Int. Thanks for pointing it out. I guess we were just following the scan function pattern. I just changed the limit argument type and made it nullable with conversion inside the implementation for both validate and scan functions. 

I'd suggest keeping the `limit` argument name, as my understanding of it is that it really does limit. The rust layer calls the argument limit and works with it like this `validate(validate_from, validate_from + limit)`. See librustacash [changelog](https://github.com/zcash/librustzcash/blob/581c686b6b10699b169d7257748c2c4ab0d3f828/zcash_client_backend/CHANGELOG.md#changed-2).

Reply:
Great, thanks for clarifying how limit is being used.  I thought it was the end of the range but your explanation makes sense.